platform/ios/src/MGLMapView.mm

```diff
@@ -1128,9 +1128,68 @@ - (void)wakeGL:(__unused NSNotification *)notification
         [self validateLocationServices];
 
         [MGLMapboxEvents pushEvent:MGLEventTypeMapLoad withAttributes:@{}];
+        
+        [self updateLocalization];
+    }
+}
+
+- (void)updateLocalization
+{
+    NSString *mapboxLanguage = [[NSUserDefaults standardUserDefaults] stringForKey:@"MGLMapboxLanguage"];
+    NSString *language = [[[NSBundle mainBundle] preferredLocalizations] firstObject];
+    NSString *preferredLanguageCode = [self preferredLanguageCode];
+    
+    if (![mapboxLanguage isEqualToString:language] && ![mapboxLanguage isEqualToString:preferredLanguageCode]) {
+        mapboxLanguage = preferredLanguageCode;
+        [[NSUserDefaults standardUserDefaults] setObject:mapboxLanguage forKey:@"MGLMapboxLanguage"];
+        [self setLabelLanguage:mapboxLanguage];
+        [[NSNotificationCenter defaultCenter] postNotificationName:NSCurrentLocaleDidChangeNotification object:nil];
```

Comment:
Posting this notification is the system’s responsibility, isn’t it? If we want to post a notification about the map’s language changing, it would need to be scoped to our framework.

Additionally, we may want to _listen_ for this notification and adjust the map’s localization accordingly.